packages/jest-haste-map/src/index.ts

```diff
@@ -739,13 +743,29 @@ class HasteMap extends EventEmitter {
     const crawlerOptions: CrawlerOptions = {
       computeSha1: options.computeSha1,
       data: hasteMap,
+      enableSymlinks: options.enableSymlinks,
       extensions: options.extensions,
       forceNodeFilesystemAPI: options.forceNodeFilesystemAPI,
       ignore,
       rootDir: options.rootDir,
       roots: options.roots,
     };
 
+    if (options.enableSymlinks) {
+      const watchmanConfigPath = path.join(options.rootDir, '.watchmanconfig');
+
+      if (existsSync(watchmanConfigPath)) {
```

Comment:
Instead of checking for the existence of a watchman config file, we should just check if Jest is configured to use it (`options.useWatchman`).

We might also consider throwing in `normalize` in `jest-config` if both options are true as they are incompatible and it would give us an even earlier error. We should still check in here though so e.g. Metro can still have the same checks.

Reply:
I noticed the watchman config option was not documented, so I just did so: 6601e921a9b2668aa70841eb09efb097f3c9d508

Reply:
Done! Nice work on the documentation :D